### PR TITLE
feat(bulma-ui): add Avatar, Avatars, and Badge components

### LINKED ISSUE #256: [Component Proposal] Avatar / Badge — user images, initials, status dots &amp; counts
## Proposed Component Name

`Avatar` (with a sibling `Avatars` group container) and `Badge` (a positioned overlay indicator).

---

## Affected Package(s)

Which package(s) should this component be added to?

- [x] bulma-ui (`@allxsmith/bestax-bulma`)
- [ ] create-bestax (`create-bestax`)
- [ ] docs (`@allxsmith/bestax-docs`)
- [ ] Other (please specify):

---

## Use Cases

User-facing UIs constantly need to represent a person or entity as a compact image — profile
menus, comment threads, member lists, chat, "assigned to" chips. Today the library has no
first-class primitive for this: people hand-roll a rounded `Image`, manually derive initials when
there's no photo, and reimplement the broken-image fallback every time.

The natural companion is a small **status/count indicator** overlaid on a corner — the online
dot, the unread count, the "3 pending" bubble. There's no primitive for that either; `Tag` is an
*inline label*, not something you position over another element.

Common cases:
- A user photo that falls back to initials (and then to a generic icon) when the image is missing
  or fails to load.
- A row of overlapping avatars for "members" with a `+5` surplus bubble when the list is long.
- An avatar with a green "online" dot, or a bell/icon with an unread **count** that caps at `99+`.
- Respect `prefers-reduced-motion` for any pulsing/processing indicator (accessibility).

---

## Proposed API

Three cooperating pieces, all CSS-driven (composed from Bulma's `image`/rounded primitives where
possible, with a small SCSS partial for the overlay positioning — no runtime animation
dependency):

```ts
// A person/entity image with graceful fallbacks.
interface AvatarProps {
  src?: string;                 // image URL; on load error, falls back to initials, then icon
  alt?: string;                 // required for meaningful images (a11y)
  name?: string;                // derives initials AND a deterministic background color when no src
  initials?: string;            // explicit initials override (else first letters of `name`)
  icon?: React.ReactNode;       // final fallback (e.g. an <Icon/>), when no src and no name
  size?: '16x16' | '24x24' | '32x32' | '48x48' | '64x64' | '96x96' | '128x128' | number; // px if number
  shape?: 'circle' | 'rounded' | 'square'; // default 'circle'
  color?: BulmaColor;           // background color for initials/icon avatars (else auto from name)
  as?: React.ElementType;       // render as 'a'/'button' for a clickable avatar (default 'figure'/'div')
  href?: string;                // when rendered as a link
  // + shared Bulma helper props (m/p, textColor, etc.)
}

// Overlapping/stacked group with a surplus bubble (mirrors the `Tags`/`Buttons` container idiom).
interface AvatarsProps {
  max?: number;                 // show first N avatars, then a "+X" surplus avatar
  size?: AvatarProps['size'];   // apply a uniform size to children
  spacing?: 'sm' | 'md' | 'lg'; // overlap amount
  // + shared Bulma helper props
}

// A small indicator overlaid on the corner of any child (or standalone).
interface BadgeProps {
  content?: number | string;    // count or short text; omit + `dot` for a plain dot
  max?: number;                 // default 99 -> renders "99+" when content exceeds it
  dot?: boolean;                // render a small dot with no content
  showZero?: boolean;           // show the badge when content === 0 (default false)
  color?: BulmaColor;           // status color (default 'danger')
  position?: 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left'; // default 'top-right'
  overlap?: 'circle' | 'square';// nudge offset for round (avatar) vs rectangular children
  pulse?: boolean;              // processing/pulse animation
  invisible?: boolean;          // hide the badge without unmounting the child
  // + shared Bulma helper props
}
```

Behavior:
- **Fallback chain is automatic:** `src` → (on error or absent) initials from `initials`/`name`
  → `icon` → a generic default. Initials avatars get a stable auto background color derived from
  `name` unless `color` is set.
- `Avatars` clamps to `max`, rendering the overflow as a single `+N` surplus avatar.
- `Badge` renders `max+` (e.g. `99+`) when a numeric `content` exceeds `max`; hides at `0` unless
  `showZero`; the `pulse` animation **no-ops under `prefers-reduced-motion: reduce`**.
- Accessibility: image avatars require `alt`; initials/icon avatars expose `role="img"` +
  `aria-label` (from `name`); a `Badge` count is announced via `aria-label`, a decorative `dot`
  is `aria-hidden`. SSR-safe (no layout shift; renders its final state on first paint).

---

## Example Usage

```tsx
import { Avatar, Avatars, Badge, Icon } from '@allxsmith/bestax-bulma';

// Photo with automatic initials fallback
<Avatar src="/users/ada.jpg" alt="Ada Lovelace" name="Ada Lovelace" size="64x64" />

// No photo -> initials on an auto color; or an icon fallback
<Avatar name="Grace Hopper" />
<Avatar icon={<Icon name="user" />} color="info" shape="rounded" />

// Overlapping member group with a surplus bubble
<Avatars max={3} size="48x48">
  {members.map(m => (
    <Avatar key={m.id} src={m.photo} alt={m.name} name={m.name} />
  ))}
</Avatars>

// Status dot and a capped unread count
<Badge dot color="success" overlap="circle">
  <Avatar src="/users/ada.jpg" alt="Ada" name="Ada" />
</Badge>

<Badge content={128} max={99} color="danger">
  <Icon name="bell" />
</Badge>
```

---

## Alternatives Considered

- **Hand-compose `Image` + `Tag`** — what people do now: a rounded `Image`, manual initials, a
  manually-positioned `Tag`. Repetitive and easy to get the fallback/positioning/a11y wrong.
- **Extend `Tag`** — `Tag` is already documented as "a label or badge," but it's an *inline*
  element. A corner **overlay indicator** (dot/count positioned over a sibling) is a distinct
  concern; overloading `Tag` with absolute positioning would muddy it. `Badge` stays focused on
  the overlay case and leaves `Tag` as the inline label.
- **Third-party component kits** — pull in extra dependencies and don't follow Bulma's
  zero-dependency, CSS-variable-driven styling. A small built-in wrapper keeps the library's
  lightweight positioning intact and themable via the existing CSS variables.

---

## Additional Context

- Confirmed absent today: no `Avatar`/`Badge` in the package; the closest primitives are `Image`
  (rounded + fixed sizes) and `Tag` (inline label). `Avatar` should reuse `Image` sizing/rounding.
- Non-stock-Bulma "extras": both need a small SCSS partial (registered in
  `src/scss/elements/_index.scss`), colors via CSS variables (never literal hex), and every class
  routed through the `ConfigProvider` prefix so `bestax-`-prefixed usage works.
- Delivers the full component anatomy: component + tests (>= 99% coverage) + Storybook stories +
  docs page + public export + regenerated skills catalog (`pnpm gen:catalog`).
- Grouping follows the existing **sibling plural container** convention (`Tags`, `Buttons`), so
  the group is `Avatars`, not `Avatar.Group`.
- If full scope is too much at once, `Avatar` + `Avatars` could land first and `Badge` follow in
  a second PR.

### PR DESCRIPTION
**Pull request by bestaxbot**
## Summary

Implements the component proposal in #256: a person/entity image primitive with automatic
fallbacks, a group container for overlapping avatars, and a corner overlay status/count
indicator.

- **`Avatar`** — renders `src` (image), falling back on load error (or absence) to initials
  (from `initials`/`name`), then a caller-provided `icon`, then a generic default icon.
  Initials get a stable auto background color derived from `name` unless `color` is set.
  Supports `size` (preset `'16x16'…'128x128'` or a pixel number), `shape`
  (`circle`/`rounded`/`square`), and renders as a link via `href`/`as`.
- **`Avatars`** — an overlapping/stacked group of `Avatar`s (mirrors the `Tags`/`Buttons`
  sibling-plural-container convention). Clamps to `max`, rendering the overflow as a single
  `+N` surplus avatar; `size` applies uniformly to every child.
- **`Badge`** — a small status/count indicator overlaid on a corner of its `children` (or
  standalone). Renders `"{max}+"` when numeric `content` exceeds `max` (default `99`), hides at
  `0` unless `showZero`, supports a plain `dot` mode, `position`/`overlap` placement, and a
  `pulse` animation that no-ops under `prefers-reduced-motion: reduce`.

All three ship with SCSS partials following the house `register-vars`/`getVar` CSS-variable
pattern (themeable, `ConfigProvider` prefix-safe), Storybook stories, Jest tests (100% statement/
branch/line coverage on all three new files), and docs pages under `docs/docs/api/components/`.

## Test plan

- [x] `pnpm --filter @allxsmith/bestax-bulma run typecheck`
- [x] `pnpm --filter @allxsmith/bestax-bulma run lint`
- [x] `pnpm --filter @allxsmith/bestax-bulma run test:coverage` (3408 tests pass; global coverage
      99.38%/99.08%/99.75%/99.59%, 100% on the three new files)
- [x] `pnpm run format:check`
- [x] `pnpm run gen:catalog` (component catalog regenerated, 85 components)
- [x] `pnpm --filter @allxsmith/bestax-bulma run build` (JS + all SCSS flavor builds, including
      `bestax-prefixed`)
- [x] Storybook production build succeeds with the new stories

Fixes #256

Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Introduced **Avatar**, **Avatars**, and **Badge** components with Bulma-compatible styling, configurable size/shape/color, interactive link support, and overflow rendering (e.g., “+N more” / max formatting).
  * Added **Avatars** spacing options and refined surplus behavior; **Badge** supports dots, positions/overlap, pulse, invisibility, and standalone mode.
* **Documentation**
  * Added/updated API pages, usage examples, accessibility guidance, theming/component-catalog entries, and guide sections.
* **Tests**
  * Added unit tests covering rendering, fallback precedence, accessibility, styling/prop behavior, and overflow/spacing edge cases.
* **Styles**
  * Added SCSS styling for avatar/avatars/badge variants and utilities.

### CODE REVIEW

File: bulma-ui/src/scss/components/_badge.scss [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  box-shadow: 0 0 0 cv.getVar('badge-ring-width') cv.getVar('badge-ring-color');
+  font-size: cv.getVar('badge-font-size');
+  line-height: 1;
+  white-space: nowrap;
+  pointer-events: none;
+
+  &.#{iv.$class-prefix}is-standalone {
+    position: static;
+    display: inline-flex;
+  }
```
[the remark this answers is not in the document]
Comment (bestaxbot) [reply bulma-ui/src/scss/components/_badge.scss]: Fixed in 3e2e44cd9d616c8267137036b25ca320d3097c33 -- confirmed the standalone badge kept the corner transform and rendered offset. One wrinkle on the suggested patch: adding transform: none inside the existing is-standalone block would not have won the cascade. .badge.is-standalone and .badge.is-top-right have equal specificity (two classes each), and the is-standalone block was declared before the corner-position rules, so is-top-right transform: translate(50%, -50%) would still have won by source order. I moved the standalone reset out to a top-level rule after the four corner-position rules and added transform: none (keeping position: static and display: inline-flex), so source order now resolves it in favor of standalone and the badge sits inline in normal flow.

File: bulma-ui/src/scss/components/_badge.scss [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    visibility: hidden;
+  }
+}
+
+// Corner positions (relative to the .badge-wrapper, unless is-standalone)
+.#{iv.$class-prefix}badge.#{iv.$class-prefix}is-top-right {
+  top: 0;
+  right: 0;
+  transform: translate(50%, -50%);
+}
```
[the remark this answers is not in the document]
Comment (bestaxbot) [reply bulma-ui/src/scss/components/_badge.scss:73]: Good catch, and your specificity analysis is exactly right -- .badge.is-standalone and .badge.is-top-right tie at two classes, so the later-declared is-top-right won the transform. Note that the proposed one-line diff (adding transform: none to the in-place is-standalone block) would not have fixed it for that same reason: the is-standalone block sits before the corner rules, so it still loses source order. I fixed it in 3e2e44cd9d616c8267137036b25ca320d3097c33 by moving the standalone reset to a top-level rule after the four corner-position rules and setting transform: none there (keeping position: static and display: inline-flex), so it now wins the cascade and the standalone badge sits inline.

File: bulma-ui/src/components/Avatars.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  const childArray = React.Children.toArray(children).filter(
+    React.isValidElement
+  ) as React.ReactElement<AvatarProps>[];
+
+  const maxCount = typeof max === 'number' && max >= 0 ? max : undefined;
+  const visibleChildren =
+    maxCount !== undefined ? childArray.slice(0, maxCount) : childArray;
+  const overflowCount =
+    maxCount !== undefined ? Math.max(childArray.length - maxCount, 0) : 0;
+
```
[the remark this answers is not in the document]
Comment (bestaxbot) [reply bulma-ui/src/components/Avatars.tsx]: Agreed, this was a real inconsistency. slice(0, 2.5) truncates to two children, but overflowCount used the raw float, so max={2.5} over three avatars rendered a +2.5 badge. I applied your fix in 872778ef91efd77e066d9143b8d2b801eb43125b: maxCount now requires Number.isInteger(max) && max >= 0, otherwise it stays undefined (no clamping), so visibleChildren and the +N label always agree. I also added a test (ignores a fractional max and clamps nothing) that renders three avatars with max={2.5} and asserts all three show and no surplus appears.

**Comment by allxsmith** [comment]
## Review — Avatar / Avatars / Badge

@bestaxbot, solid foundation: all three components follow the house patterns correctly (`useBulmaClasses` + `usePrefixedClassNames` + `classNames` join, `ConfigProvider` prefix support in both TSX and SCSS, `register-vars`/`getVar` theming, prefix-asserting tests, autodocs stories, catalog regenerated). The items below need to change before this merges.

### Bugs

- [ ] **Standalone `Badge` renders offset.** `.badge.is-standalone` resets `position`, but the component always emits a position class, and `is-top-right`'s `transform: translate(50%, -50%)` applies to statically-positioned elements too — so `<Badge content={5} color="info" />` (the docs "Standalone" example and story) is shifted up-right by half its own size. Fix: don't emit `is-{position}`/`is-overlap-*` when there are no `children` (optionally also `transform: none` on `.is-standalone`).
- [ ] **`Badge` sends pass-through props to the wrong element — and sometimes drops them.** With `children`, the rendered root is the `badge-wrapper` span, but `className`, the Bulma helper classes, and `{...rest}` all go to the inner pill, which is `pointer-events: none` — so `onClick`/mouse handlers can never fire, and spacing helpers (`m="3"`) move the floating pill instead of the component. When the badge hides (`content={0}` without `showZero`), `rest`/`className` are dropped entirely (`data-testid`/`id` vanish). House convention is root receives them (see `Tag`, `Tooltip`, `Dropdown`). Restructure: the root (wrapper when `children`, pill when standalone) gets `className` + helpers + rest; always render the wrapper when `children` exist; add a `badgeClassName` slot prop for the pill (like `Tooltip`'s `tooltipClassName`, but merge it with plain `classNames` — don't prefix it). Keep both `usePrefixedClassNames` calls hoisted above the early returns (Rules of Hooks).
- [ ] **`Avatar` misses image errors that fire before hydration.** The fallback relies solely on React's `onError`; in SSR/static pages an image that fails before hydration never triggers it, leaving a permanently broken ``. Other popular libraries detect this on mount — add a `ref` + `useEffect` treating `img.complete && img.naturalWidth === 0` as an error.
- [ ] **`getInitialsFromName` breaks on astral-plane characters** (emoji, some CJK): it indexes UTF-16 code units and can produce half-surrogate initials. Use `Array.from(...)` to take code points.
- [ ] **`Avatar` forwards `href` unconditionally**, so `<Avatar as="div" href="...">` emits a stray attribute, and `target`/`rel` aren't typed. Follow `Button`: declare `target`/`rel` props and forward `href/target/rel` only when rendering an `a` (or a custom `as` component).

### Missing functionality we should have (users coming from other popular libraries will expect these)

- [ ] **`Avatars` needs a non-overlapping mode.** The group can currently only overlap. Add `spaced?: boolean` → `.avatars.is-spaced` with `gap` (house precedent: `Rate`'s `spaced` prop in `_rate.scss`), reusing the sm/md/lg spacing tokens for the gap size and dropping the ring shadow in spaced mode.
- [ ] **Compound API: `Avatars.Avatar`.** Attach the existing `Avatar` as a static — `React.FC<AvatarsProps> & { Avatar: typeof Avatar }` plus `Avatars.Avatar = Avatar`, exactly the `Steps.Step`/`Tabs.Tab` pattern. `Avatar` stays a separate export too.
- [ ] **Uniform `shape` on `Avatars`.** A group of square avatars currently gets a circular `+N` bubble. Add `shape?: AvatarProps['shape']` cloned like `size` — and use a conditional spread so unset group props don't override children (the current `size ?? child.props.size` always overrides via the `cloneElement` merge, making the fallback branch unreachable).
- [ ] **Never render a pointless `+1`.** When exactly one child overflows `max`, the `+1` bubble occupies exactly the slot the hidden avatar would — render the avatar instead (`overflow === 1 ? 0 : overflow`).
- [ ] **Numeric `spacing`.** Accept a pixel number in addition to `'sm' | 'md' | 'lg'`; with the single spacing CSS variable (SCSS section below) this is one inline custom-property style.
- [ ] **`imageProps` on `Avatar`.** There's no way to reach the inner `` (`srcSet`, `sizes`, `crossOrigin`, `referrerPolicy`, observing `onError`…). Add `imageProps?: React.ImgHTMLAttributes<HTMLImageElement>`, chaining its `onError` with the internal handler.
- [ ] **`Badge` `content` should accept `React.ReactNode`**, not just `number | string` (e.g. a tiny icon in the pill). Keep the `max`/`showZero` logic gated on `typeof content === 'number'`.

Nice-to-haves for a follow-up issue rather than this PR: a `total` prop (server-known count driving `+N` without rendering every child) and a `renderSurplus` render-prop for custom overflow content.

### SCSS conformance (register-vars pattern)

- [ ] `$avatars-overlap-sm/md/lg` are consumed via raw interpolation, so the overlap isn't runtime-themeable. Register ONE `avatars-spacing` var (default md) and re-register it under `.is-spacing-sm`/`.is-spacing-lg` (see `_sidebar.scss`'s `is-narrow`); consume as `margin-left: calc(-1 * cv.getVar('avatars-spacing'))` — spaced mode's `gap` then reads the same var.
- [ ] `$badge-inset-circle` and `$badge-animation-duration` are used directly — add both to `register-vars` and consume via `cv.getVar` (`_dialog.scss` registers its animation duration).
- [ ] `$badge-inset-square` is declared and never used — delete it.
- [ ] `is-surplus` is emitted by `Avatars` but styled nowhere, and the surplus bubble's `color="light"` doesn't flip in dark mode. Style `.avatars .avatar.is-surplus` with scheme-aware tokens (`background`/`text`) and drop `color="light"` from the TSX.
- [ ] `border-radius: 9999px` for `is-circle` → `cv.getVar('radius-rounded')` (as in `_taginput.scss`, `_slider.scss`, `_switch.scss`).

### Docs pages

- [ ] avatar.md "Photo with automatic fallback": the gray bulma.io placeholder is indistinguishable from a broken image — use `https://github.com/allxsmith.png`. The section also never demonstrates the fallback it's named for — add a broken-`src` example (`https://example.invalid/missing.jpg` → initials), like the existing story.
- [ ] Glued examples: avatar.md Initials/Shapes/Sizes render avatars flush together inside `<Block>` — wrap them in the new `<Avatars spaced>` container. avatars.md "Spacing" renders two groups flush on one line so they read as a single malformed six-avatar group — stack them. badge.md "Positions" needs a flex/gap wrapper: each badge translates outside its anchor and lands on the neighboring one. (The stories already use gap wrappers; the docs ports dropped them.)
- [ ] badge.md: give the `Icon` anchors an `ariaLabel` (icon.md convention) and add the real-world examples this kind of component is usually shown with: unread mail (envelope icon), badge-on-`Button` (cart count), an `invisible` toggle with `useState`, `pulse`, a color row, and `overlap="circle"` vs `"square"` side by side. `pulse` and `invisible` are documented props with zero examples today.
- [ ] Structure: one sentence of prose under every example heading; Title Case headings; end pages with `## Related Components` + `## Additional Resources` like the rest of the docs (not `## See also`); avatars.md is missing an `## Accessibility` section entirely.
- [ ] badge.md Accessibility: add the guidance that screen readers need full context on the anchor itself (e.g. `aria-label="Inbox, 4 unread messages"`), not a bare number in a `role="status"` span.
- [ ] badge.md Overview: one line distinguishing `Tag` (inline pill — we already have that) from `Badge` (positioned overlay).
- [ ] Props tables after the changes above: add `badgeClassName`, `target`/`rel` (link-only forwarding note), `shape`/`spaced`, reword `spacing` ("overlap amount — or gap when `spaced`"), and note that `position`/`overlap` are ignored for standalone badges. Also document that the `visibility="invisible"` helper hides the whole wrapper while the `invisible` prop hides only the pill.

### Tests / stories

- [ ] Missing behavioral tests: Badge `content === max` boundary; wrapper keeps `data-testid`/`id` and `onClick` fires when the badge is hidden; Avatar `as="button"` (the button half of `isInteractive` is never exercised — a 99%-branch-coverage risk); Avatars `max={0}` and negative `max`; the surplus bubble's accessible name (`"2 more"`).
- [ ] Stories: add Badge `invisible`, string content (`"NEW"`), standalone dot; Avatars `spaced` and `shape`; argTypes should carry `description` fields (house style, cf. `Dialog.stories.tsx`); Avatars size options stop at `64x64` while Avatar supports up to `128x128`.

### Skills (same-PR rule in CLAUDE.md)

- [ ] `skills/bestax-theming/references/themeable-components.md`: add Avatar (`color`, 10 values + `size`) and Badge (`color`, 10 values).
- [ ] `skills/bestax-theming/references/css-variables.md`: add the new `--bulma-avatar-*` / `--bulma-avatars-*` / `--bulma-badge-*` variables, including the ones newly registered above.

### Not this PR (context for reviewers)

The "little square" under the badge examples in the deployed docs preview is **not** an example bug: the docs build strips Font Awesome's `@font-face` rules site-wide (the CSS minifier's unused-font pass can't trace FA7's CSS-variable font-family indirection), so every FA icon in every live docs preview renders as a missing-glyph box — production included. That needs a docs-infra fix (disable `discardUnused` via `configurePostCss`) and is being handled separately; don't "fix" the icon examples in response to it.

**Comment by allxsmith** [comment]
@bestaxbot you agreed with the Bugs section of my review — go ahead and apply those five fixes now.

**Comment by allxsmith** [comment]
@bestaxbot what about addressing:

Missing functionality we should have (users coming from other popular libraries will expect these)
SCSS conformance (register-vars pattern)
Docs / Pages
Tests / Stories
Skills (same-PR rule in CLAUDE.md)

I don't see that you addressed those issues in the review

**Comment by allxsmith** [comment]
@bestaxbot let's try the Missing functionality section again: the non-overlapping `spaced` mode, uniform `shape`, and numeric `spacing` on Avatars; the `Avatars.Avatar` compound static; the no-pointless-`+1` overflow behavior; `imageProps` on Avatar; and ReactNode `content` on Badge — each with its stories, tests, and docs updates.

Important: commit and push after each cohesive piece lands rather than batching everything into one final commit, so nothing is lost if you hit your turn budget. If you can't finish everything, reply with exactly what's done and what's left.

**Comment by allxsmith** [comment]
Okay now let's address the remaining sections from my review

SCSS conformance (register-vars pattern)
Docs / Pages
Tests / Stories
Skills (same-PR rule in CLAUDE.md)

**Comment by allxsmith** [comment]
Noticed that the new component is missing from several docs pages

- missing from the intro components page /docs/guides/library/components
- missing from the intro home page

**Comment by allxsmith** [comment]
@bestaxbot did you see my latest comment?

**Comment by allxsmith** [comment]
On the home docs page you have the components listed under the bulma section, it should be under beyond Bulma as avatar and badge.   Don't list avatars

**Comment by allxsmith** [comment]
In the docs, i see inline style that should be removed and replaced with our components and use our props.

For instance, on https://67d95204.bestax.pages.dev/docs/guides/library/components#badge

Replace:
<div style={{ display: 'flex', gap: '2rem' }}>

with something like the Block component, with flex helpers

Please checkout the storybook stories to look for inline style related to these components: Avatar, Avatars, Badge

**Comment by allxsmith** [comment]
I still see inline style:

https://56ebcdc0.bestax.pages.dev/docs/guides/library/components#badge

<div style={{ display: 'flex', gap: '2rem' }}>
  <Badge content={4} color="danger">
    <Avatar name="Ada Lovelace" />
  </Badge>
  <Badge content={128} max={99} color="info">
    <Avatar name="Grace Hopper" shape="square" />
  </Badge>
  <Badge dot color="success" overlap="circle">
    <Avatar name="Katherine Johnson" />
  </Badge>
</div>
